docs/HowToFAQ/Errors-CommandLine.md

```diff
@@ -1,24 +1,231 @@
+command is invoked or are absolute file paths. In particular, this
+instance of this error message identifies non-.dfy files that still
+have permitted extensions (according to the target platform) but
+cannot be found in the file system.
+
+## **Error: Command-line argument '_arg_' is neither a recognized option nor a filename with a supported extension (_ext_).** {#cli_bad_option}
+
+<!-- %check-cli -->
+```bash
+dafny resolve --zzz
+```
+
+The named command-line argument is either a misspelled option or a filename
+in an unrecognized form (e.g., with no filename extension).
+An invalid option with a filename as value can look like either one, such as `--out:a.doo`.
+
+## **Error: Command-line argument '_arg_' is neither a recognized option nor a filename with a supported extension (_ext_).** {#cli_bad_option_legacy}
+
+<!-- %check-cli -->
+```bash
+dafny -zzz
+```
+
+The named command-line argument is either a misspelled option or a filename
+in an unrecognized form (e.g., with no filename extension).
+
+## **Error: '_name_': Filename extension '_ext_' is not supported. Input files must be Dafny programs (.dfy) or supported auxiliary files (_ext_)** {#cli_bad_extension}
+
+<!-- %check-cli -->
+```bash
+dafny resolve z.zzz
+```
+
+Dafny programs are in files with a `.dfy` extension.
+A Dafny program may be combined with other files appropriate to the
+target programming language being used, such as `.dll` files for C#
+or `.java` or `.jar` files for Java. The stated extension is not
+recognized.
+
+## **Error: The command-line contains no .dfy files** {#cli-no-dafny-file_2}
+
+<!-- TODO - this appears to be no longer reachable -->
+
+## **Error: The command-line contains no .dfy files** {#cli-no-dafny-folder}
+
+<!-- TODO - this appears to be no longer reachable -->
+
+## **Error: Only one .dfy file can be specified for testing** {#cli-test-generation}
+
+<!-- %no-check -->
+```bash
+dafny generate-tests block x.dfy y.dfy
+```
+
+The test-generation tool is still experimental.
+However, it is expected to take just a single `.dfy` file as input.
+
+
+## **Error: ModelView file must be specified when attempting counterexample extraction** {#cli-counterexample}
+
+<!-- %check-cli -->
+```bash
+dafny /extractCounterexample testsource/test1.dfy
+```
+
+When requesting a counterexample to be generated, some additional options must be specified:
+a model view file using `/mv:<file>` and
+`/proverOpt:O:model.completion=true` and
+`/proverOpt:O:model_compress=false` (for z3 version < 4.8.7) or
+`/proverOpt:O:model.compact=false `(for z3 version >= 4.8.7).
+
+The model view file is an arbitrary non-existent file that is created and used as a temporary file
+during the creation of the counterexamples.
+
+## **Error: Feature not supported for this compilation target: _feature_** {#f_unsupported_feature}
+
+<!-- %check-run %exit 3 %options -t:cpp --unicode-char=true -->
+```dafny
+method m() {}
+```
```

Comment:
This error message seems self-explanatory, but is it intentional that there is no additional description here?

Reply:
Added a description.